raiden/transfer/channel.py

```diff
@@ -1184,10 +1184,22 @@ def get_max_withdraw_amount(
     sender: NettingChannelEndState, receiver: NettingChannelEndState
 ) -> WithdrawAmount:
     """Calculates the maximum "total_withdraw_amount" for a channel.
-    This will leave the channel without funds, when
+    This will leave the channel without funds, when this is withdrawn from the contract,
+    or pending as offchain-withdraw.
 
     """
-    return WithdrawAmount(sender.total_withdraw + get_balance(sender, receiver))
+    sender_transferred_amount = 0
+    receiver_transferred_amount = 0
+
+    if sender.balance_proof:
+        sender_transferred_amount = sender.balance_proof.transferred_amount
+
+    if receiver.balance_proof:
+        receiver_transferred_amount = receiver.balance_proof.transferred_amount
+
+    return WithdrawAmount(
+        sender.contract_balance - sender_transferred_amount + receiver_transferred_amount
+    )
```

Comment:
I would like to see the changes to `transfer/channel.py` moved into a separate commit, since this does not seem to be a part of the straightforward test-split topic.

Reply:
Done.